src/main/java/org/systemexception/sparktutorial/pojo/SparkContext.java

```diff
@@ -15,33 +17,52 @@
  */
 public class SparkContext {
 
-	private final SparkConf sparkConf;
 	private final JavaSparkContext sparkContext;
-	private final Logger logger;
 
 	public SparkContext() {
-		sparkConf = new SparkConf().setAppName("org.systemexception.sparktutorial").setMaster("local");
+		SparkConf sparkConf = new SparkConf().setAppName("org.systemexception.sparktutorial").setMaster("local");
+		PropertyConfigurator.configure(SparkContext.class.getClassLoader().getResource("spark_log4j.properties"));
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [In J2EE, getClassLoader() might not work as expected.  Use Thread.currentThread().getContextClassLoader() instead.](https://www.codacy.com/app/leo_4/spark-tutorial/file/1878983176/issues/source?bid=3087676&fileBranchId=3088177#l24)